src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.AbstractPriorityProcessor.cs

```diff
@@ -36,6 +37,8 @@ private abstract class AbstractPriorityProcessor : GlobalOperationAwareIdleProce
                     {
                         _gate = new object();
                         _lazyAnalyzers = lazyAnalyzers;
+                        _lazyAnalyzersForActiveDocumentChanged = new Lazy<ImmutableArray<IIncrementalAnalyzer>>(
+                            () => Analyzers.WhereAsArray(a => a.IsDocumentAnalysisDependentOnItBeingActiveDocumentOrNot));
```

Comment:
i don't love that thsi feels racy (two computed properties dependent on each other).  A cleaner and safer way to do this imo is to make _lazyAnalzyers a `Lazy<(ImmutableArray<IIncrementalAnalyzer> allAnalzyers, ImmutableArray<IIncrementalAnalyzer> activeDocumentChangedAnalyzers)>`.  Then just lazily compute both at once and store them, and have the properties return .item1 or .item2.

Reply:
This code has now been removed.